pootle/apps/pootle_store/unit/results.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from itertools import groupby
+
+from django.core.urlresolvers import reverse
+
+from pootle.core.url_helpers import split_pootle_path
+from pootle.i18n.gettext import language_dir
+from pootle_store.models import FUZZY
+from pootle_store.templatetags.store_tags import (
+    pluralize_source, pluralize_target)
+from pootle_store.unit.proxy import UnitProxy
+
+
+class UnitResult(UnitProxy):
+
+    @property
+    def pootle_path(self):
+        return self.unit["store__pootle_path"]
+
+    @property
+    def project_code(self):
+        return self.unit["store__translation_project__project__code"]
+
+    @property
+    def project_style(self):
+        return self.unit[
+            "store__translation_project__project__checkstyle"]
+
+    @property
+    def source_dir(self):
+        return language_dir(self.source_lang)
+
+    @property
+    def source_lang(self):
+        return self.unit[
+            "store__translation_project__project__source_language__code"]
+
+    @property
+    def target_dir(self):
+        return language_dir(self.target_lang)
+
+    @property
+    def target_lang(self):
+        return self.unit[
+            "store__translation_project__language__code"]
+
+    @property
+    def translate_url(self):
+        store_url = u''.join(
+            [reverse("pootle-tp-store-translate",
+                     args=split_pootle_path(self.pootle_path))])
+        return (
+            "%s%s"
+            % (store_url,
+               '#unit=%s' % unicode(self.id)))
+
+
+class StoreResults(object):
+
+    def __init__(self, units):
+        self.units = units
+
+    @property
+    def data(self):
+        meta = None
+        units_list = []
+
+        for unit in iter(self.units):
+            unit = UnitResult(unit)
+            if meta is None:
```

Comment:
Doing this for all units looks like a bit inefficient, so better do it once before `for`.